.github/workflows/gradle.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         uses: actions/setup-java@v4
         with:
           distribution: 'corretto'
-          java-version: '17'
+          java-version: '21'
```

Comment:
What's the reason for changing the Java version used for the regular CI runs? I don't think the Java version here should have any impact on the new CI job that you've added.

Also, why is the Java version important? Does something fail if we use Java 17? If so, that should be fixed: Specimin should run under Java 17+.